Tools/Temp/ChangeEditingContextTool.cs

```diff
@@ -0,0 +1,85 @@
+﻿#if UNITY_EDITOR
+using System;
+using System.Collections.Generic;
+using UnityEditor.Experimental.EditorVR.Core;
+using UnityEditor.Experimental.EditorVR.Utilities;
+using UnityEngine;
+using UnityEngine.InputNew;
+
+namespace UnityEditor.Experimental.EditorVR.Tools
+{
+	[MainMenuItem("Change Context", "Create", "Push and pop editing contexts")]
+	//[MainMenuItem(false)]
+	sealed class ChangeEditingContextTool : MonoBehaviour, ITool, ICustomActionMap, ISetEditingContext, ITooltip, 
+		ISetTooltipVisibility, ITooltipPlacement, IUsesRayOrigin
+	{
+		[SerializeField]
+		ActionMap m_ActionMap;
+
+		int m_CurrentContextIndex;
+		Transform m_TooltipTransform;
+
+		public List<IEditingContext> allContexts { private get; set; }
+
+		public ActionMap actionMap { get { return m_ActionMap; } }
+
+		public string tooltipText { get; private set; }
+		public Transform tooltipTarget { get { return m_TooltipTransform; } }
+		public Transform tooltipSource { get { return rayOrigin; } }
+		public TextAlignment tooltipAlignment { get { return TextAlignment.Center; } }
+
+		public Transform rayOrigin { get; set; }
+
+		void Awake()
+		{
+			m_TooltipTransform = ObjectUtils.CreateEmptyGameObject(null, transform).transform;
+		}
+
+		void OnDestroy()
+		{
+			if (gameObject.activeInHierarchy)
+				this.HideTooltip(this);
+
+			ObjectUtils.Destroy(m_TooltipTransform.gameObject);
```

Comment:
Destroying the toolTip transform via selecting the tool in the main menu, then selecting it again in the main menu (without using the tool), will throw an exception in MathUtilsExt, SetTransformOffset();